kiarash/TS/TS: OOP/src/index.ts

```diff
@@ -0,0 +1,61 @@
+type Branch = "architecture" | "clean-code" | "languages";
+
+abstract class Concept {
+  constructor(protected branch: string, private name: string) {}
+
+  getName() {
+    return this.name;
+  }
+
+  getBranch() {
+    return this.branch;
+  }
+}
+
+class DevConcept extends Concept {
+  constructor(branch: Branch) {
+    super(branch, "DevConcept");
+  }
+}
+
+class PhilosophicalConcept extends Concept {
+  private reasoning_list: string[] = [];
+  constructor(branch: string, reasoning?: string[], getBranch?: () => string) {
+    super(branch, "PhilosophicalConcept");
+    if (reasoning) {
+      this.reasoning_list = [...this.reasoning_list, ...reasoning];
+    }
+    if (getBranch) {
+      this.getBranch = getBranch;
+    }
+  }
+
+  addReasoning(reason: string): void {
```

Comment:
Please be consistent about return types: Either write return types for all methods, or remove it from all (except the ones that need explicit types).